src/idl2json/src/untyped_conversion.rs

```diff
@@ -1,26 +1,29 @@
 use candid::parser::value::IDLValue;
 use serde_json::value::Value as JsonValue;
 
+use crate::{bytes::convert_bytes, Idl2JsonOptions};
+
 /// Converts a candid IDLValue to a serde JsonValue, without type information.
-pub fn idl2json(idl: &IDLValue) -> JsonValue {
+pub fn idl2json(idl: &IDLValue, options: &Idl2JsonOptions) -> JsonValue {
     match idl {
         IDLValue::Bool(bool) => JsonValue::Bool(*bool),
         IDLValue::Null => JsonValue::Null,
         IDLValue::Text(s) => JsonValue::String(s.clone()),
         IDLValue::Number(s) => JsonValue::String(s.clone()), // Unspecified number type
-        IDLValue::Float64(f) => {
-            JsonValue::Number(serde_json::Number::from_f64(*f).expect("A float's a float"))
-        }
-        IDLValue::Opt(value) => JsonValue::Array(vec![idl2json(value)]),
-        IDLValue::Vec(value) => JsonValue::Array(value.iter().map(idl2json).collect()),
+        IDLValue::Float64(f) => serde_json::Number::from_f64(*f)
+            .map(JsonValue::Number)
+            .unwrap_or_else(|| JsonValue::String("NaN".to_string())),
+        IDLValue::Opt(value) => JsonValue::Array(vec![idl2json(value, options)]),
+        IDLValue::Vec(value) => convert_bytes(value, options)
```

Comment:
Is a `IDLValue::Vec` always a vector of bytes? Can't it be a vector of records for example?

Reply:
Yes it can.  If it is a vector of bytes, `convert_bytes` will succeed.  If it's anything else, it will fail and go to `convert_non_bytes_array()` on the line below.

Reply:
Ok, understood. Thanks for the explanations.

And this looks such a simple library an functionality...